woocommerce/woocommerce-gateway-stripe
# Use payment information payment method types when creating setup intents

Reported in https://github.com/woocommerce/woocommerce-gateway-stripe/issues/3046#issuecomment-2041976774

## Changes proposed in this Pull Request:

On stores with Stripe Link enabled, when a customer attempted to sign up to a free-trial subscription they would receive the following error: 

<p align="center">
<img width="710" alt="Screenshot 2024-07-24 at 11 13 10 AM" src="https://github.com/user-attachments/assets/9e88c87d-5ec5-4ce6-920a-71726c13b086">
</p> 

This error is caused because when we generate the setup intent request args, the `payment_method_types` property would only include the selected type and so would be set to `[ 'card' ]`. This would lead to the failure because `link` isn't included as a eligible payment type. 

This PR fixes that by making sure we use `$payment_information['payment_method_types']` which is generated here and includes `link` when Stripe Link is enabled (see [code here](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/8.5.0/includes/payment-methods/class-wc-stripe-upe-payment-gateway.php#L2428-L2451)). This is consistent with how the payment intent flow works. See here: [`create_and_confirm_payment_intent()`](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/8.5.0/includes/class-wc-stripe-intent-controller.php#L718). 

## Testing instructions

1. Enable Stripe Link in the Stripe payment method settings. 
2. Enable WooCommerce Subscriptions. 
3. Create a subscription product with a free trial.
   - This is important because we need the checkout to be free for the setup-intent flow. 
4. On checkout make sure you're logged into Link by clicking on the Link login the email field. 
5. Attempt to complete the signup. 
   - On `develop` you will see the error I mentioned above. 
   - On this branch the sign up will be successful. 





---


**Post merge**
